### PR TITLE
UHC  refactor backend

### PR DESCRIPTION
**Pull request by joshzarrabi**
## Description

Moves all calculations needed to show UHC metrics to the backend. The data API now serves all metrics exactly as they will be seen on the frontend. This includes generating an API for national level data.

Also does some basic refactoring of the UHC code to make it more readble.

## Motivation and Context

See #1476

## How has this been tested?

Yes, added lots of unit tests and did lot of spot checks. The UI should look exactly the same as it currently does but it should just be alot faster.

## Types of changes

- Refactor (code improvement that doesn't affect UI)

### CODE REVIEW

File: python/datasources/uhc.py [hunk 1]
```diff
@@ -145,150 +140,208 @@ def upload_to_gcs(self, _, **attrs):
...
+                gcs_to_bq_util.add_df_to_bq(
+                    breakdown_df, dataset, table_name, column_types=column_types)
 
-            if std_col.RACE_INCLUDES_HISPANIC_COL in breakdown_df.columns:
-                column_types[std_col.RACE_INCLUDES_HISPANIC_COL] = 'BOOL'
 
-            gcs_to_bq_util.add_df_to_bq(
-                breakdown_df, dataset, breakdown, column_types=column_types)
+def parse_raw_data(df, breakdown):
+    """Parses the raw UHC data into a datatframe with the column names and datatypes
```
Comment (benhammondmusic) [inline python/datasources/uhc.py:173]: typo

File: python/tests/datasources/test_uhc_brfss.py [hunk 2]
```diff
@@ -1,85 +1,341 @@
...
+    "non_medical_rx_opioid_use_per_100k": float,
+    "non_medical_drug_use_per_100k": float,
+    "excessive_drinking_per_100k": float,
+    "preventable_hospitalizations_per_100k": float,
+    "avoided_care_per_100k": float,
+    "chronic_kidney_disease_per_100k": float,
+    "cardiovascular_diseases_per_100k": float,
+    "asthma_per_100k": float,
+    "voter_participation_per_100k": float,
+    'brfss_population_pct': str
```
Comment (benhammondmusic) [inline python/tests/datasources/test_uhc_brfss.py:83]: is this supposed to be `str` or should it be `float`

File: python/tests/datasources/test_uhc_brfss.py [hunk 3]
```diff
@@ -1,85 +1,341 @@
...
+    expected_df = pd.read_json(
+        GOLDEN_DATA_SEX, dtype=expected_dtype)
+
+    assert_frame_equal(
+        mock_bq.call_args_list[2].args[0], expected_df, check_like=True)
+
+
+# For the national level we only need to make sure that we are making the
+# correct call to bigquery to get population data, so that is all we need to
+# test. There is no need to maintrain GOLDEN files for this, as there is no
```
Comment (benhammondmusic) [inline python/tests/datasources/test_uhc_brfss.py:231]: typo "maintain"

File: python/datasources/uhc.py [hunk 4]
```diff
@@ -145,150 +140,208 @@ def upload_to_gcs(self, _, **attrs):
...
-                                               ] = matched_row['Value'].values[0]
-                                # convert from % to per 100k
-                                else:
-                                    output_row[UHC_DETERMINANTS[determinant]
-                                               ] = matched_row['Value'].values[0] * 1000
+def estimate_total(row, condition_name_per_100k):
+    """Returns an estimate of the total number of people with a given condition.
 
-                output.append(output_row)
+       sample_per_100k: a percentage of people in a demographic with a given condition, represented as a per 100k
```
Comment (benhammondmusic) [inline python/datasources/uhc.py:338]: I'm not sure I follow this comment; do you need to update the arguments in the doc string?

File: python/datasources/uhc.py [hunk 5]
```diff
@@ -145,150 +140,208 @@ def upload_to_gcs(self, _, **attrs):
...
+    if pd.isna(row[condition_name_per_100k]) or \
+        pd.isna(row[std_col.POPULATION_COL]) or \
+            int(row[std_col.POPULATION_COL]) == 0:
 
-        if breakdown == std_col.RACE_OR_HISPANIC_COL:
-            std_col.add_race_columns_from_category_id(output_df)
+        return None
 
-        return output_df
+    return round((float(row[condition_name_per_100k]) / 100000) * float(row[std_col.POPULATION_COL]))
```
Comment (benhammondmusic) [inline python/datasources/uhc.py:347]: should `condition_name_per_100k` be something like `condition_raw_count` instead? I'm not clear why you need to divide a per100k rate by 100k again.

**Review by benhammondmusic — APPROVED** [review]
Awesome. Hopefully we see some nice speed improvements w all this

File: python/tests/datasources/test_uhc_brfss.py (hunk 2 above)
Comment (joshzarrabi) [reply python/tests/datasources/test_uhc_brfss.py:83]: I'll try it with float

File: python/datasources/uhc.py (hunk 4 above)
Comment (joshzarrabi) [reply python/datasources/uhc.py:338]: it should say `per_100k of people` good catch

File: python/datasources/uhc.py (hunk 5 above)
Comment (joshzarrabi) [reply python/datasources/uhc.py:347]: you divide the `per_100k` number to get the raw ratio, which you then multiply by the population.

**Pull request merged**